File: solutions/_1408_string_matching_in_an_array/solution.py

from typing import List


class Solution:
    def stringMatching(self, words: List[str]) -> List[str]:
        result = []
        for i, word in enumerate(words):
            for j, other in enumerate(words):
                if i != j and word in other:
                    result.append(word)
                    break
        return result
  
# https://leetcode.com/problems/string-matching-in-an-array/

File: solutions/_1408_string_matching_in_an_array/solution.php

<?php

class Solution {

    /**
     * @param String[] $words
     * @return String[]
     */
    function stringMatching($words) {
        $result = [];
        foreach ($words as $i => $word) {
            foreach ($words as $j => $other) {
                if ($i !== $j && strpos($other, $word) !== false) {
                    $result[] = $word;
                    break;
                }
            }
        }
        return $result;
    }
}

/*
https://leetcode.com/problems/string-matching-in-an-array/
*/
